meta-oe/recipes-connectivity/hiba/hiba_git.bb

```diff
@@ -0,0 +1,40 @@
+
+SUMMARY = "Google Host Identity Based Authorization"
+DESCRIPTION = "HIBA is a system built on top of regular OpenSSH \
+  certificate-based authentication that allows to manage flexible \
+  authorization of principals on pools of target hosts without the need to \
+  push customized authorized_users files periodically."
+
+S = "${WORKDIR}/git"
+
+LICENSE = "BSD"
+LIC_FILES_CHKSUM = "file://${S}/LICENSE;md5=7e161abe9a4585310725dd86c28d1ae2"
+
+SRC_URI = "git://github.com/google/hiba;protocol=https;branch=main"
+SRCREV = "3199239c6a0a89dbd8f155204f44798c78195ab0"
+
+PV = "1.0+git${SRCPV}"
+
+inherit autotools
+
+DEPENDS = "openssl openssh zlib"
+RDEPENDS:${PN} = "bash"
+
+def get_oeconf(d):
+  import glob
+  workdir = d.getVar("WORKDIR")
+  openssh_paths = glob.glob("{}/../../openssh/*/openssh-*".format(workdir))
```

Comment:
I don't believe you can rely on directories above WORKDIR within the recipe.

Reply:
oh dang, I am not too sure what else to use then.

Reply:
I'm guessing you'll need to modify the openssh recipe instead to make the files that you need available

Reply:
where should I install the files? William mentioned that we shouldn't use the `STAGING_DIR_NATIVE` which I thought is the sysroot. (maybe not)

Reply:
if you are using gut of a package ( e.g. openssh ) without respecting what the component expect their users to use via headers and libs then I would urge you to take a look at this component and try to either use what openssh exposes or make it expose what you need. Yocto builds in recipe specific sysroots, they dont see anything outside that sysroot during build. As such we have shared state and this will break horribly if their are dependencies outside its own sysroot.

Reply:
um, so it sounds like we could not build `hiba` in bitbake since it depends on stuff that is not exposed.

I guess this means that the only way to do it cleanly is to have the `hiba` as a submodule of `openssh-portable` to enforce that all the required stuff are available... but then that will be dependent on the openssh release to get the new changes... which is not ideal.

@blundere what do you think? Seems like the way it's done is not proper... and I have ran out of ideas.

Reply:
as a submod of openssh seems would work and sounds right too. You can use a `PACKAGECONFIG` knob then to control turning it on/off

Reply:
As I understand it, we need to be modifying the OpenSSH recipe to expose the headers, libssh.a and libopenbsd-compat.a to the rest of the world. There are 2 ways:

1. I can contact the owner pf OpenSSH to see if they would accept a patch to support it direclty via make their make install  
2. We can have OpenSSH inherit lib_package class and start exposing manually in a install hook the files we want.

I am also happy with HIBA being a submod of OpenSSH as this is in line with reality

Reply:
I think the issues right now is that each sysroot in bitbake is based on recipes, so we can't get reliability get the build folder of another repo. So I think `lib_package` will hit the same issue.

Yeah, seems like a submodule in openssh is the best move and maybe we don't use the official release of openssh for `hiba` until they are available. Build from SRC_URI and such.

Reply:
Thanks @blunderer's feedback, I have updated the latest recipe changes. It seems like I just need to install the headers + lib*.so for it to work. I thought we needed the object files and was hesitant to do so.

It should be better now.

Reply:
its better than before. However, I think it still needs modifications in openssl as I grok. If it does, then get those changes into openssh recipe first, the recipe needs to build with standard openssh provided by oe-core for it to be in any open layers.

Reply:
sounds good. Thanks make sense. We can follow up on that before getting this submitted.

If Google/OpenBMC needs it earlier, we will support it within our meta-layer until the openssh changes are ready. and then migrate it here.

Thanks for the help and suggestions.

Reply:
There should be no modification to openssl required since openssl library and includes are installed in the sysroot already and available for HIBA to link against in a standard way.

Reply:
Will move the recipe change to poky for now.

Reply:
Poky review in https://git.yoctoproject.org/poky-contrib/commit/?h=wltu/install_openssh_files&id=c8441295669fd11c0fe3f88f943d3afcd5fc172c. PTAL.

Reply:
Oops. I updated to https://git.yoctoproject.org/poky-contrib/?h=wltu%2Finstall_internal_openssh_files and cc you guys in the email. I am not sure how to get more attention to the change or if I did anything wrong.

Any tips?